### PR TITLE
Add #inspectVersions method to list installed packages

### PR DESCRIPTION
**Pull request by jackfranklin**
Returns a promise that resolves to an array of:

```
[
  { dep: 'npm:foo', version: '1.1.1', linked: false },
  ...
]
```

@guybedford how does this look? I still don't quite understand all the details of this so I might have missed something.

### CODE REVIEW

File: lib/install.js [hunk 1]
```diff
@@ -805,7 +805,35 @@ function showVersions(forks) {
...
   });
 
-  versions = alphabetize(versions);
+  return {
+    versions: alphabetize(versions),
+    linkedVersions: linkedVersions
+  };
+}
+
+function inspectVersions(forks) {
```
Comment (jackfranklin) [inline lib/install.js]: this `fork` argument actually isn't being used - not sure if I should be or not. A bit unclear on how this affects the output.

**Comment by guybedford** [comment]
Thanks, glad to see the small diff principles in action!

We just need to consider the output data here a little bit carefully. Strictly, just taking the most recent version is incorrect because I could have installed a library to 1.0.1, while a dependency of mine uses version 3.5.0 of that same library. The version we're interested in is more likely to be the version I personally installed than the dependency.

The most general data structure to represent what I have installed is exactly the combination of map config and package map config - that is I have primary resolutions in the first map config, and secondary resolutions that trace out of the dependency maps from there.

Perhaps it would help to first understand your use case here a little better to see what this output is for? Then lets try to put something together that captures the tree metadata as generally as possible?

**Comment by jackfranklin** [comment]
Sure :)

The main motivation is that I was working on a `jspm-outdated` script that would parse your package.json and list:

- the semver range for each jspm package
- the current installed version
- the latest version from github / npm

So that you could easily update and keep things up to that.

**Comment by guybedford** [comment]
Perhaps then you could work with an output like:

```javascript
{
  "primary": {
    "bootstrap": {
      "target": "github:twbs/bootstrap@^3.0.0",
      "resolution": "github:twbs/bootstrap@3.0.1",
      "linked": false
    }
  },
  "secondary": {
    "github:twbs/bootstrap@3.0.1": {
      "jquery": {
        "target": "github:components/jquery@^2.0.1",
        "resolution": "github:components/jquery@2.3.4",
        "linked": false
      }
    }
  }
}
```

Where users have added custom maps, the resolution might be a local map like `./local/path.js` in those cases.

Let me know your thoughts on the above?

**Comment by jackfranklin** [comment]
That looks pretty good :+1:

**Comment by guybedford** [comment]
That this would likely no longer fall under the `jspm inspect` function though. It may just be an API function for analysis, perhaps along the lines of `getInstallTree` or something similar.

**Comment by jackfranklin** [comment]
That's fine - this only started out being under `inspect` because I was stealing the code ! I'm happy for it to go wherever you think makes sense :)

**Pull request closed without merging**